src/authHook.ts

```diff
@@ -228,3 +225,42 @@ export const getPluginConfigStartingCwd = (configuration: Configuration): Portab
   const isDlx = !!configuration.projectCwd?.endsWith(`dlx-${process.pid}`)
   return isDlx ? npath.toPortablePath(process.cwd()) : configuration.startingCwd
 }
+
+// Dependabot relies on this env variable so it's existence points to the fact
+// that we are running such an environment. See
+// https://github.com/dependabot/dependabot-core/blob/23de1c7583117bd15f26ab33482383ed57208f14/updater/lib/dependabot/environment.rb#L6
+//
+// Dependabot doesn't support passing environment variables. We can only pass the
+// token for the registry in a special dependabot.yaml config file. So this plugin
+// will never work for it, just skip the token calculation and configure it separately.
+const isRunningInDependabot = (): boolean => process.env.DEPENDABOT_JOB_ID !== undefined
+
+// In some environments you may want to use an existing auth token instead of fetching a new one
+// This is an escape hatch for that case
+const useExistingAuthtoken = (): boolean => process.env._YARN_PLUGIN_AWS_CODEARTIFACT_DISABLE !== undefined
+
+const skipPlugin = (): boolean => isRunningInDependabot() || useExistingAuthtoken()
+
+export const SKIP_PLUGIN_ERROR = 'CODEARTIFACT_AUTH_TOKEN is not set; cannot use _YARN_PLUGIN_AWS_CODEARTIFACT_DISABLE'
+export const DEPENDABOT_DUMMY_TOKEN = 'dummy-token'
+export const skipPluginToken = () => {
+  if (isRunningInDependabot()) {
+    // return a dummy header to prevent `No authentication configured for request`
+    // see https://github.com/yarnpkg/berry/blob/ad8c95d3bd597966b4669d5fff13a95deab550af/packages/plugin-npm/sources/npmHttpUtils.ts#L384
+    // the dependabot proxy will replace the dummy header with a valid one before calling the registry
+    return `Bearer ${DEPENDABOT_DUMMY_TOKEN}`
+  }
+  if (useExistingAuthtoken()) {
+    const existingAuthToken = process.env.CODEARTIFACT_AUTH_TOKEN
+
+    if (!existingAuthToken) throw new Error(SKIP_PLUGIN_ERROR)
+
+    if (process.env._YARN_PLUGIN_AWS_CODEARTIFACT_TESTING) {
+      console.log(`_YARN_PLUGIN_AWS_CODEARTIFACT_DISABLE: Use passed in token: ${existingAuthToken}`)
+    }
+
+    return `Bearer ${existingAuthToken}`
+  }
+
+  throw Error('This function should not be called if the plugin is not skipped')
```

Comment:
this shouldn't be possible to hit, but it is a guardrail in case anyone ever makes it possible.